File: src/hw1/task5/Task5.java
package hw1.task5;

import java.util.ArrayList;

public class Task5 {
    public static void main(String[] args) {
        File file1 = new File("file1");
        File file2 = new File("file2.png");
        File file3 = new File("file3.jpg");
        File file4 = new File("file1.webp");
        File file5 = new File("file1.bz2");

        Folder folder1 = new Folder("folder1", file1, file2);
        Folder folder2 = new Folder("folder2", file3, file4, file5);

        Folder root = new Folder("root", folder1, folder2);

        System.out.println(file4.getPath());

        ArrayList<FileSystemNode> folder1Children = folder1.getChildren();
        File fileWithExtension = (File) folder1Children.get(1);
        File fileWithoutExtension = (File) folder1Children.get(0);
        System.out.println(fileWithExtension.getExtension());
        System.out.println(fileWithoutExtension.getExtension());
    }
}

File: src/hw2/task5/Task5.kt
package hw2.task5

object Task5 {

    @JvmStatic
    fun main(args: Array<String>) {
        val file1 = File(null, "file1.txt")
        val file2 = File(null, "file2.png")
        val file3 = File(null, "file3.jpg")
        val file4 = File(null, "file1.webp")
        val file5 = File(null, "file1.bz2")

        val folder1 = Folder(null, "folder1", file1, file2)
        val folder2 = Folder(null, "folder2", file3, file4, file5)

        val root = Folder(null, "root", folder1, folder2)

        println(file4.getPath())

        val folder1Children: Array<out FileSystemNode> = folder1.getChildren()
        val file = folder1Children[1] as File
        println(file.getExtension())
    }
}